### PR TITLE
5300: Add initial transaction menu

### PR DESCRIPTION
**Pull request by X0nic**
project_with_djeva_-_madeline

It is hard to read the url in the url bar, so here is it decoded in Chrome.

la_base_united_states_internal

### CODE REVIEW

**Comment by X0nic** [comment]
Hey @linesarefuzzy, this is basically done. Just missing the date fields right now. How do we want to format the dates for the other system?

**Comment by linesarefuzzy** [comment]
I'd say YYYY-MM-DD. That's pretty much an international standard, for computers at least.

**Comment by X0nic** [comment]
Looks like if I just include Date's but not DateTime's everything comes out properly.
internal_labase_org_loanschedule_php_loanid_1460_loan_5bamount_5d_160000_0_loan_5bcurrency_id_5d_1_loan_5bdivision_id_5d_13_loan_5bid_5d_1460_loan_5blength_months_5d_6_loan_5bname_5d_project_with_grafica_22_de_mayo_loan_5borganization_id_5d

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -53,4 +53,49 @@ def render_index_grid(grid)
...
+  end
+
+  # Returns the schedule url, as well as addtional loan and org information.
+  # This information maybe be required, if the loan does not exist in the old system.
+  def labase_schedule_url(loan:)
+    labase_url('LoanSchedule.php', loan, 'LoanID': loan.id)
+  end
+
+  # Takes the page_url merges labase values, with loan values, and returns properly formatted URI
+  def labase_url(page_url, loan, labase_query_values)
```
Comment (X0nic) [inline app/helpers/application_helper.rb]: `labase` for the accounting system. Is this a good name?

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -53,4 +53,49 @@ def render_index_grid(grid)
     no_records = grid.current_page_records.length < 1
     render "admin/common/grid", no_records: no_records, grid: grid
   end
+
+  # Returns the new disbursment url, as well as addtional loan and org information.
+  # This information maybe be required, if the loan does not exist in the old system.
+  def labase_new_disbursment_url(loan:)
```
Comment (X0nic) [inline app/helpers/application_helper.rb]: The url that gets generated is pretty long and verbose, I opted to not include an example in the URL. They look like this: `http://internal.labase.org/LoanSchedule.php?LoanID=1460&loan%5Bamount%5D=160000.0&loan%5Bcurrency_id%5D=1&loan%5Bdivision_id%5D=13&loan%5Bid%5D=1460&loan%5Blength_months%5D=6&loan%5Bname%5D=Project%20with%20Gr%C3%A1fica%2022%20de%20Mayo&loan%5Borganization_id%5D=454&loan%5Borganization_snapshot_id%5D=1325&loan%5Bprimary_agent_id%5D=255&loan%5Bpublic_level_value%5D=hidden&loan%5Brate%5D=32.5&loan%5Bsecondary_agent_id%5D=145&loan%5Bsigning_date%5D=2016-10-03&loan%5Bstatus_value%5D=prospective&organization%5Balias%5D=La%20Cabril&organization%5Bcity%5D=Avellaneda&organization%5Bcountry_id%5D=1&organization%5Bdivision_id%5D=2&organization%5Bemail%5D=quintanacarlos68%40gmail.com&organization%5Bid%5D=454&organization%5Bindustry%5D=Gr%C3%A1fica&organization%5Blegal_name%5D=Cooperativa%20de%20Trabajo%20Gr%C3%A1fica%2022%20de%20MayoLimitada&organization%5Bname%5D=Gr%C3%A1fica%2022%20de%20Mayo&organization%5Bneighborhood%5D=&organization%5Bprimary_phone%5D=4208%201150&organization%5Breferral_source%5D=Red%20Gr%C3%A1fica&organization%5Bsector%5D=Gr%C3%A1fica&organization%5Bstate%5D=&organization%5Bstreet_address%5D=Rivadavia%20700%20%28Avellaneda%29&organization%5Btax_no%5D=30714015458`

And like this when decoded by chrome:
internal_labase_org_loanschedule_php_loanid_1460_loan_5bamount_5d_160000_0_loan_5bcurrency_id_5d_1_loan_5bdivision_id_5d_13_loan_5bid_5d_1460_loan_5blength_months_5d_6_loan_5bname_5d_project_with_grafica_22_de_mayo_loan_5borganization_id_5d

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -53,4 +53,49 @@ def render_index_grid(grid)
     no_records = grid.current_page_records.length < 1
     render "admin/common/grid", no_records: no_records, grid: grid
   end
+
```
Comment (linesarefuzzy) [inline app/helpers/application_helper.rb]: This is great stuff but it feels wrong to put it in application_helper. How about a new helper file? "old_system_helper"? Or maybe it even belongs in its own class? Thoughts? I bet @hooverlunch has some, if he wants to comment.

In either case, I'd make the transaction type ids and the base_uri into constants.

File: app/helpers/application_helper.rb [hunk 4]
```diff
@@ -53,4 +53,49 @@ def render_index_grid(grid)
...
+  # =>
+  # {"loan[id]"=>1364,
+  #  "loan[amount]"=>#<BigDecimal:7feb498d5658,'0.1E5',9(18)>,
+  #  "loan[currency_id]"=>2 ... }
+  def to_query_values(object)
+    allowed_attribs = object.attributes.select do |_key, value|
+      [Fixnum, String, BigDecimal, Date].any? { |type| value.is_a?(type) }
+    end
+
+    allowed_attribs.inject({}) { |acc, (key, value)| acc.update("#{object.class.to_s.downcase}[#{key}]" => value.to_s) }
```
Comment (linesarefuzzy) [inline app/helpers/application_helper.rb]: This is very concise, but a bit dense and not easy to tell what's going on. I think I'd prefer an `each`. But it's not a big deal.

File: app/helpers/application_helper.rb (hunk 3 above)
Comment (X0nic) [inline app/helpers/application_helper.rb]: Yeah I agree, we should move this. Seems like overkill to put in its own class, but I would be fine with either way.

Comment (hooverlunch) [inline app/helpers/application_helper.rb]: I'm fine with just a separate helper in this case since the code will eventually go away anyway.

Comment (X0nic) [inline app/helpers/application_helper.rb]: Good point, probably not worth spending too much time on it, since it will be going away.

**Pull request merged**